File: Python/Arrays/Binary_Search.py
# Implementing Binary Search

# Binary search has the primary condition of the array to be sorted

# Input the sorted array 
arr = []
size = int(input("Enter the size of the array: "))
for i in range(size):
    val = int(input("Enter the value: "))
    arr.append(val)

# Input the element to be found
n = int(input("Enter a element to be found: "))

# Creating a function for Binary Search
def BinarySearch(arr,n):
    l = 0
    r = len(arr) - 1
    while(l<=r):
        mid = len(arr)//2
        if(arr[mid]<n):
            l = mid + 1
        elif(arr[mid]>n):
            r = mid - 1
        else:
            return mid
    return -1

# Printing the element
result = BinarySearch(arr,n)
if(result==-1):
    print("Element not found")
else:
    print("Element found at index", result)

File: Java/Arrays/BinarySearch.java
import java.util.*;

class Search{
	int binary_search(int[] arr, int element) {
		int start = 0;
		int end = arr.length-1;
		while(start<=end) {
			int mid = (start+end)/2;
			if(element==arr[mid]) {
				return mid;
			}
			else if(element<arr[mid]) {
				end = mid-1;
			}
			else {
				start = mid+1;
			}
		}
		return -1;
	}
}

public class BinarySearch {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int[] arr = {1,3,5,8,9,12};
		System.out.println("Enter the element you want to search: ");
		int element = sc.nextInt();
		Search bs = new Search();
		int result = bs.binary_search(arr, element);
        if (result == -1) {
        	System.out.println("Element not found in the array.");
        } 
        else {
        	System.out.println("The position of the element is: " + result);
        }
	}

}
